docs/manual/scala/guide/services/ServiceDescriptors.md

```diff
@@ -0,0 +1,115 @@
+Dynamic path parameters are extracted from the path by declaring dynamic parts in the path.  These are prefixed with a colon, for example, a path of `/order/:id` has a dynamic part called `id`. Lagom will extract this parameter from the path, and pass it to the service call method. In order to convert it to the type accepted by the method, Lagom will use an implicitly provided [`PathParamSerializer`](api/com/lightbend/lagom/scaladsl/api/deser/PathParamSerializer.html).  Lagom includes many `PathParamSerializer`'s out of the box, such as for `String`, `Long`, `Int`, `Boolean` and `UUID`.  Here's an example of extracting a `long` parameter from the path and passing it to a service call:
+
+@[call-long-id](code/ServiceDescriptors.scala)
+
+Note that this time we're using an [eta-expanded](http://scala-lang.org/files/archive/spec/2.11/06-expressions.html#method-values) reference to the method. This is because the method takes a parameter.
+
+Multiple parameters can of course be extracted out, these will be passed to your service call method in the order they are extracted from the URL:
+
+@[call-complex-id](code/ServiceDescriptors.scala)
+
+Query string parameters can also be extracted from the path, using a `&` separated list after a `?` at the end of the path.  For example, the following service call uses query string parameters to implement paging:
+
+@[call-query-string-parameters](code/ServiceDescriptors.scala)
+
+When you use `call`, `namedCall` or `pathCall`, if Lagom maps that down to REST, Lagom will make a best effort attempt to map it down to REST in a semantic fashion. So for example, if there is a request message it will use the `POST` method, whereas if there's none it will use `GET`.
+
+### REST identifiers
+
+The final type of identifier is a REST identifier. REST identifiers are designed to be used when creating semantic REST APIs.  They use both a path, as with the path based identifier, and a request method, to identify them.  They can be configured using the [`restCall`](api/com/lightbend/lagom/scaladsl/api/Service$.html#restCall[Request,Response]\(Method,String,ScalaMethodServiceCall[Request,Response]\)\(MessageSerializer[Request,_],MessageSerializer[Response,_]\):Call[Request,Response]) method:
+
+@[call-rest](code/ServiceDescriptors.scala)
+
+## Messages
+
+Every service call in Lagom has a request message type and a response message type.  When the request or response message isn't used, the `akka.NotUsed` can be used in their place.  Request and response message types fall into two categories, strict and streamed.
+
+### Strict messages
+
+A strict message is a single message that can be represented by a simple Java object.  The message will be buffered into memory, and then parsed, for example, as json.  When both message types are strict, the call is said to be a synchronous call, that is, a request is sent and received, then a response is sent and received.  The caller and callee have synchronized in their communication.
```

Comment:
This is probably copied over from the Java, but it would be good to be consistent about capitalizing JSON as we do in the next paragraph.